script/prerender/fileUtils.ts

```diff
@@ -42,3 +47,34 @@ export const createDiskCache = <K extends string, V>(prefix: string): Cache<K, V
     },
   };
 };
+
+let s3Client: S3Client | undefined;
+
+async function writeS3File(contentType: string, filepath: string, contents: string) {
+  let basePath = assertDefined(process.env.PUBLIC_URL, 'PUBLIC_URL environment variable not set');
+  if (basePath[0] === '/') { basePath = basePath.slice(1); }
+  const key = `${basePath}${filepath}`;
+
+  if (!s3Client) {
+    // We explicitly pass credentials to the S3 Client only once during initialization
+    // so it'll not try to load them over and over
+
+    console.log('Fetching container credentials');
+    const credentials = await fromContainerMetadata();
+
+    console.log('Initializing S3 client');
+    s3Client = new S3Client({ credentials, region: process.env.BUCKET_REGION });
+  }
+
+  console.log(`Writing s3 file: /${key}`);
+  return s3Client.send(new PutObjectCommand({
+    Body: contents,
+    Bucket: process.env.BUCKET_NAME,
+    CacheControl: 'max-age=0',
+    ContentType: contentType,
+    Key: key,
+  }));
+}
+
+export const writeS3HtmlFile = writeS3File.bind(null, 'text/html');
```

Comment:
i'd have done this with function scope, with a semi curried function, rather than bind and messing around with the whole `this` context

i think that is just personal preference though i don't know if there is necessarily anything wrong with doing it this way

Reply:
MDN docs claim that bind(null) doesn't change `this` but I guess they lied:
`what()` is a function that logs `this`.

```
> obj.what = what
[Function: what]
> obj.what()
{ what: [Function: what] }
undefined
> obj.what = what.bind(null)
[Function: bound what]
> obj.what()
<ref *1> Object [global] {
...
```

I'll rewrite this not to use bind()...

Reply:
The have been rewritten not to use `bind()`.